pkg/controller/node/node_controller_test.go

```diff
@@ -1012,6 +1013,42 @@ func TestShouldDoNothing(t *testing.T) {
 	f.run(getKey(mcp, t))
 }
 
+func TestSortNodeList(t *testing.T) {
+	old_node := newNode("node-6", "v1", "v1")
+	old_node.CreationTimestamp = metav1.NewTime(time.Unix(0, 42))
+
+	newer_node := newNode("node-6", "v1", "v1")
+	newer_node.CreationTimestamp = metav1.NewTime(time.Unix(200, 30))
+
+	nodes := []*corev1.Node{
+		newNodeWithLabel("node-0", "v1", "v1", map[string]string{"topology.kubernetes.io/zone": "ZZZ"}),
```

Comment:
I think this is a good start, but let's also unit test 3 zones and also test how maxUnavailable plays into this per above?

Reply:
Ah yes I had 3 already but was testing with a duplicate for my own ref. can readd